p22/src/bin/song.rs

```diff
@@ -0,0 +1,9 @@
+use p22::song::Song;
+
+pub fn main() {
+    let full_song = Song::new().compute_all();
+
+    for day in full_song {
+        println!("{day}")
```

Comment:
This part here -

another question: in case I want to inject/fake/mock stderr/stdout, can I do directly or do I need dependency injection?

The src/bin/song.rs is not properly tested and I don't know how to.

I've read this answer: https://stackoverflow.com/questions/28370126/how-can-i-test-stdin-and-stdout

Reply:
AFAIK there are not straightforward solutions for this. You don't need to test it in this exercise and can just verify that printed text is correct manually.